qemu/tests/qemu_guest_agent.py

```diff
@@ -1901,10 +1905,34 @@ def gagent_check_fsfreeze(self, test, params, env):
             self.gagent.fsthaw(check_status=False)
 
         self._action_before_fsfreeze()
-        error_context.context("Freeze the FS.", logging.info)
-        self.gagent.fsfreeze()
+        error_context.context("Freeze the FS when fsfreeze_list is"
+                              " %s and mountpoints is %s." %
+                              (fsfreeze_list, mountpoints), logging.info)
+        self.gagent.fsfreeze(fsfreeze_list=fsfreeze_list,
+                             mountpoints=mountpoints)
         try:
-            self._action_after_fsfreeze(write_cmd, write_cmd_timeout)
+            if fsfreeze_list:
+                if not mountpoints:
+                    # fsfreeze-list without mountpoints, will freeze all fs.
+                    mountpoints = mountpoints
```

Comment:
Does this meaningful to assign a variant to itself?

Reply:
yes, will modify.